Modules/Core/Sources/Core/Buffer/Buffer.swift

```diff
@@ -105,6 +108,40 @@ public struct Buffer : RandomAccessCollection {
         }
         
     }
+
+    public func withUnsafeBufferPointer<R>(_ body: (UnsafeBufferPointer<Byte>) throws -> R) rethrows -> R {
+        return try bytes.withUnsafeBufferPointer(body)
+    }
+
+    public mutating func withUnsafeMutableBufferPointer<R>(_ body: (inout UnsafeMutableBufferPointer<Byte>) throws -> R) rethrows -> R {
+        return try bytes.withUnsafeMutableBufferPointer(body)
+    }
+}
+
+extension UnsafeBufferPointer {
+    public init() {
+        self.init(start: nil, count: 0)
+    }
+
+    public init(capacity: Int) {
+        let pointer = UnsafeMutablePointer<Element>.allocate(capacity: capacity)
```

Comment:
What is the point in having this on `UnsafeBufferPointer` as you can never actually write to the pointer?